pkg/capture/capturetypes/packet.go

```diff
@@ -41,23 +41,18 @@ const (
 
 )
 
-// GPPacket stores all relevant packet details for a flow
-type GPPacket struct {
-	// packet size
-	NumBytes uint32
-
-	// packet inbound or outbound on interface
-	DirInbound bool
-
-	// flag to easily determine if a packet is IPv4 or IPv6
-	IsIPv4 bool
-
-	// packet descriptors / hashes
-	EPHash        EPHash
-	EPHashReverse EPHash
-
-	// Auxilliary data (protocol dependent) for further analysis
-	AuxInfo byte
+// EPHash is a typedef that allows us to replace the type of hash
+type EPHash [37]byte
+
+// Reverse calculates the reverse of an EPHash (i.e. source / destination switched)
+func (h EPHash) Reverse() (rev EPHash) {
```

Comment:
New method for on-demand reversal of the hash. Since it's an array the whole thing is kept on the stack (unless of course a function explicitly lets it escape to the heap).